src/axios/axios.js

```diff
@@ -0,0 +1,116 @@
+const methods = {
+    GET: "GET",
+    PUT: "PUT",
+    HEAD: "HEAD",
+    POST: "POST",
+    PATCH: "PATCH",
+    DELETE: "DELETE",
+    OPTIONS: "OPTIONS",
+};
+
+const axios = function (config) {
+    const { url, method, data, params } = config;
+    switch (method) {
+        case methods.POST:
+            return axios.post(url, data, config);
+        case methods.DELETE:
+            return axios.delete(url, config);
+        case methods.PUT:
+            return axios.put(url, data, config);
+        case methods.PATCH:
+            return axios.patch(url, data, config);
+        case methods.HEAD:
+            return axios.head(url, config);
+        case methods.OPTIONS:
+            return axios.options(url, config);
+        case methods.GET:
+        default:
+            return axios.get(url, params, config);
+    }
+};
+
+axios.request = function (...args) {
+    return axios(...args);
+};
+
+axios.get = (url, params = {}, config = {}) => {
+    config.method = methods.GET;
+    config.headers["Content-Type"] =
```

Comment:
can we extract headers part outside its common for all function, similar with 2nd param handling
you can create new function 
`config = formatPayload(methods.GET); // it will return config`

Reply:
if you are able to create formatPayload function then your function will look like this
```js
axios.post = (url, data = {}, config = {}) => {
    const { url, config } = formatPayload({url, method: methods.GET, data, config});
    return fetch(url, config);
};
```
all function will be two line
so dynamically create all functions using methods
Object.keys(methods).forEach(method=>{
axios[method.toLowerCase] = () => {
// ...
}
})